solver/src/driver.rs

```diff
@@ -285,23 +288,78 @@ impl Driver {
         }
     }
 
-    // Record metric with the amount of orders that were matched but not settled in this runloop (effectively queued for the next one)
-    // Should help us to identify how much we can save by parallelizing execution.
-    fn report_matched_but_unsettled_orders(
+    // Record metric with surplus achieved in winning settlement
+    // vs that which was unrealized in other feasible solutions.
+    fn surplus_report(
         &self,
-        submitted: &Settlement,
-        all: impl Iterator<Item = RatedSettlement>,
+        submitted: &[Trade],
+        all_trades: &[Trade],
+        prices: &HashMap<H160, BigRational>,
     ) {
-        let submitted: HashSet<_> = submitted
-            .trades()
+        let submitted_surplus: HashMap<_, _> = submitted
+            .iter()
+            .map(|trade| {
+                (
+                    trade.order.order_meta_data.uid,
+                    trade
+                        .surplus(
+                            &prices[&trade.order.order_creation.sell_token],
+                            &prices[&trade.order.order_creation.buy_token],
+                        )
+                        .expect("I can't imagine submitting a trade with None surplus"),
```

Comment:
Following the calls to surplus around a bit I think this can happen because we have one function that turns None into 0 in Settlement::total_surplus and we might settle a settlement with 0 surplus.

Reply:
I only glanced at the code so I'm not sure. I think it could happen on some overflow. It is probably a very rare case so ignoring it for this metric is fine but we shouldn't panic.

Reply:
Right... so ignoring this case here without panicing sounds like `unwrap_or(ZERO)` to me.